startup/92-optimization.py

```diff
@@ -237,20 +239,16 @@ def _watch_function(self, *args, **kwargs):
 # RE(bp.fly([hf]))
 
 
-def calc_velocity(motors, dists, velocity_limits, max_velocity=None):
-    #TODO: finish testing/debugging this)
-
-    # TODO: update this to handle a max velocity
+def calc_velocity(motors, dists, velocity_limits, max_velocity=None, min_velocity=None):
     ret_vels = []
-    # if any([velocity_limits[i][1] == 0 for i in range(len(velocity_limits))]):
-    #     # make sure all motors have upper limits > 0
-    #     raise ValueError("Upper velocity limit must be > 0 for every motor.")
+    # check that max_velocity is not None if at least 1 motor doesn't have upper velocity limit
+    if any([velocity_limits[i]['high'] == 0 for i in range(len(velocity_limits))]) and max_velocity is None:
+        raise ValueError('max_velocity must be set if there is at least 1 motor without upper velocity limit')
```

Comment:
It would be useful to also show which motors are affected. All motors having an unset `max_velocity` should be reported to allow a user to address the misconfigured motors in one shot.